File: solutions/oscillation.cpp
#include <iostream>

int main() {
    int steps;
    float pos, target, step, range;

    steps = 0;
    std::cin >> pos >> target >> step >> range;
    while (true) {
        if (abs(pos-target) <= range) {
            std::cout << steps << std::endl;
            break;
        }
        if (pos < target) {
            while (pos < target-range) {
                pos += step;
                steps++;
            }
        } else {
            while (pos > target+range) {
                pos -= step;
                steps++;
            }
        }
        step /= 2;
    }
}

File: solutions/oscillation.py
pos, target, step, target_range = map(int, input().strip().split())
steps = 0
while True:
    if abs(pos-target) <= target_range:
        print(steps)
        break
    if pos < target:
        while pos < target-target_range:
            pos += step
            steps += 1
    else:
        while pos > target+target_range:
            pos -= step
            steps += 1
    step /= 2
